### PR TITLE
Allow build_barback 0.4.x from build_test

### PR DESCRIPTION
**Pull request by natebosch**
Currently there exists a version lock trying to upgrade to build 0.10

### CODE REVIEW

**Comment by natebosch** [comment]
So this depends on getting build 0.10 for the tests to pass - I *think* that should be OK. The `lib/` stuff should all be backwards compatible still.

**Comment by natebosch** [comment]
Travis won't pass for this one until we get some of these new versions published.

Will separately look at the formatting issue.

File: build/pubspec.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ dependencies:
   glob: ^1.1.0
 
 dev_dependencies:
-  build_barback: ^0.2.0
+  build_barback: ^0.4.0
```
Comment (jakemac53) [inline build/pubspec.yaml]: is 0.2.0 still compatible? If so I would just change it to be >=0.2.0 <0.5.0

Comment (natebosch) [inline build/pubspec.yaml]: No it's not, build_barback has a public implementation of the Resolver interface which saw breaking changes.

**Pull request merged**